lib/client.js

```diff
@@ -177,32 +179,15 @@ class AssetComputeClient extends EventEmitter {
         }
 
         // Register I/O event type and journal, emit events
-        const { journal } = await this.assetCompute.register();
-        const eventEmitter = new AssetComputeEventEmitter({
-            ...this.options,
-            accessToken: this.accessToken,
-            org: this.integration.technicalAccount.org,
-            journal
-        });
-
-        this.eventEmitter = eventEmitter;
-
-        // event forwarding
-        const self = this;
-        this.eventEmitter.on('rendition_created', event => {
-            if (getAssetComputeClientId(event) === self.id) {
-                completePendingRendition(self);
-                self.emit("rendition_created", event);
-            }
-        });
-        this.eventEmitter.on('rendition_failed', event => {
-            if (getAssetComputeClientId(event) === self.id) {
-                completePendingRendition(self);
-                self.emit("rendition_failed", event);
-            }
-        });
-        this.eventEmitter.on('error', error => self.emit("error", error));
+        const response = await this.assetCompute.register();
+        this.journal = response.journal;
         this._registered = true;
+
+        if (this.eventEmitter) {
+            await this.eventEmitter.close();
+            this.eventEmitter = undefined; // new journal, must reset eventEmitter
```

Comment:
Nit-pick: If it was used before, better reset it to null (undefined is more to mean it was never used before).